platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -167,14 +168,26 @@ def __mount_onie_fs(self):
             self.__umount_onie_fs()
 
         cmd = "fdisk -l | grep 'ONIE boot' | awk '{print $1}'"
-        fs_path = subprocess.check_output(cmd,
-                                          stderr=subprocess.STDOUT,
-                                          shell=True,
-                                          universal_newlines=True).rstrip('\n')
+        cmd1 = ['fdisk', '-l']
+        cmd2 = ['grep', 'ONIE boot']
+        cmd3 = ['awk', '{print $1}']
+        with subprocess.Popen(cmd1, universal_newlines=True, stdout=subprocess.PIPE) as p1:
```

Comment:
>with subprocess.Popen(cmd1, universal_newlines=True, stdout=subprocess.PIPE) as p1: [](http://example.com/codeflow?start=8&length=83)

Possible to move to common library, maybe the 2 cmd function could be extended to multiple cmds.